realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -1187,6 +1271,25 @@ private void emitInsertOrUpdateMethod(JavaWriter writer) throws IOException {
                         .endControlFlow()
                         .emitEmptyLine();
 
+            } else if (Utils.isRealmValueList(field)) {
+                final String genericType = Utils.getGenericTypeQualifiedName(field);
+                final TypeMirror elementTypeMirror = Utils.getRealmListElementTypeMirror(field);
+                writer
+                        .emitEmptyLine()
+                        .emitStatement("OsList %1$sOsList = new OsList(table.getUncheckedRow(rowIndex), columnInfo.%1$sIndex)", fieldName)
+                        .emitStatement("%1$sOsList.removeAll()", fieldName)
```

Comment:
removeAll just removes them, but don't delete them, right?

Reply:
Right! `OsList` as another method `deleteAll()` that deletes linked object from Realm.

`removeAll()` and `deleteAll()` are the same if the list is primitive list.